app/models/machine.rb

```diff
@@ -1,5 +1,16 @@
 class Machine < ApplicationRecord
   validates_presence_of :location
-  
+
   belongs_to :owner
+
+  has_many :snack_machines
+  has_many :snacks, through: :snack_machines
+
+  def average_snack_price
+    self.snacks.average(:price)
+  end
+
+  def snack_count
+    self.snacks.count
+  end
```

Comment:
These look great. No need to use self in your instance methods since it's assumed, but doesn't hurt to have it.